### PR TITLE
PIM-5096: Add XLSX quick export

### PR DESCRIPTION
**Pull request by SamirBoulil**
Adds an XLSX quick export button to the product grid.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | Yes
| Changelog updated                 | Yes
| Review and 2 GTM                  | Yes
| Micro Demo to the PO (Story only) | Yes
| Migration script                  |  Yes
| Tech Doc                          | -

### CODE REVIEW

File: features/mass-action/quick_export.feature [hunk 1]
```diff
@@ -33,3 +33,21 @@ Feature: Quick export many products from datagrid
...
+    Then I should have 1 new notification
+    And I should see notification:
+      | type    | message                                                      |
+      | success | Quick Export Quick export XLSX product quick export finished |
+    Then I go on the last executed job resume of "xlsx_product_quick_export"
+    And I should see "COMPLETED"
+    And the path of the exported file of "xlsx_product_quick_export" should be "/tmp/products_export_en_US_tablet.xlsx"
+    And exported xlsx file of "xlsx_product_quick_export" should contain:
+      | sku      | 123 | categories        | color | description-en_US-tablet | enabled | family   | groups | lace_color | manufacturer | name-en_US    | price-EUR | price-USD | rating | side_view | size | top_view | weather_conditions |
+      | boots    | aaa | winter_collection | black |                          | 1       | boots    |        |            |              | Amazing boots | 20        | 25        |        |           | 40   |          |                    |
```
Comment (SamirBoulil) [inline features/mass-action/quick_export.feature]: For some reason, the double quotes appearing in `"Amazing boots"` the CSV file disappear in xlsx.. idk if it's relevant.

File: features/localization/mass-action/quick_export.feature [hunk 2]
```diff
@@ -35,3 +35,23 @@ Feature: Quick export many products with localized attributes from datagrid
     sandals;summer_collection;red;;;1;sandals;;;;Sandals;5;5;75;;;40;;;0,50;GRAM
     pump;summer_collection;blue;;;1;;;;;Pump;15;20;;;;41;;;;
     """
+
+  Scenario: Successfully quick export XLSX products with localized attributes
+    Given I am on the products page
+    And I select rows boots, sneakers, sandals and pump
```
Comment (willy-ahva) [inline features/localization/mass-action/quick_export.feature]: When

File: features/localization/mass-action/quick_export.feature [hunk 3]
```diff
@@ -35,3 +35,23 @@ Feature: Quick export many products with localized attributes from datagrid
...
+    Then I press "XLSX (tous les attributs)" on the "Export rapide" dropdown button
```
Comment (willy-ahva) [inline features/localization/mass-action/quick_export.feature]: And

File: features/localization/mass-action/quick_export.feature [hunk 4]
```diff
@@ -35,3 +35,23 @@ Feature: Quick export many products with localized attributes from datagrid
...
+    And I wait for the "xlsx_product_quick_export" quick export to finish
+    When I am on the dashboard page
+    Then I should have 1 new notification
+    And I should see notification:
+      | type    | message                                                   |
+      | success | Exportation rapide XLSX product quick export est terminée |
+    Then I go on the last executed job resume of "xlsx_product_quick_export"
```
Comment (willy-ahva) [inline features/localization/mass-action/quick_export.feature]: When

File: features/localization/mass-action/quick_export.feature [hunk 5]
```diff
@@ -35,3 +35,23 @@ Feature: Quick export many products with localized attributes from datagrid
...
+    And I should see "TERMINÉ"
```
Comment (willy-ahva) [inline features/localization/mass-action/quick_export.feature]: Then

File: features/mass-action/quick_export.feature [hunk 6]
```diff
@@ -33,3 +33,21 @@ Feature: Quick export many products from datagrid
     boots;aaa;winter_collection;black;;1;boots;;;;"Amazing boots";20;25;;;40;;
     sneakers;bbb;summer_collection;white;;1;sneakers;;;;Sneakers;50;60;;;42;;
     """
+
+  Scenario: Successfully quick export products as a XSLX file
```
Comment (willy-ahva) [inline features/mass-action/quick_export.feature]: same comments this scenario

**Comment by willy-ahva** [comment]
GTM after comments fixed

File: src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php [hunk 7]
```diff
@@ -124,4 +124,20 @@ public function setWithHeader($withHeader)
     {
         $this->withHeader = $withHeader;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setConfiguration(array $config)
```
Comment (juliensnz) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: why do we need that ?

Comment (SamirBoulil) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: I added it, in order for the job to replace parameters in the filepath. eg, locale and channel placeholders.

**Comment by juliensnz** [comment]
GTM after comment responded :)

File: features/mass-action/quick_export.feature (hunk 6 above)
Comment (SamirBoulil) [inline features/mass-action/quick_export.feature]: Fixed

File: src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php (hunk 7 above)
Comment (juliensnz) [inline src/Pim/Component/Connector/Writer/File/XlsxProductWriter.php]: good :)

**Comment by jjanvier** [comment]
GTM

File: upgrades/schema/Version_1_6_201603100000_batch_jobs.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+     * @param Schema $schema
+     */
+    public function up(Schema $schema)
+    {
+        $this->addSql(<<<SQL
+            INSERT INTO akeneo_batch_job_instance
+                (`code`, `label`, `alias`, `status`, `connector`, `rawConfiguration`, `type`)
+            VALUES
+                ('xlsx_product_export', 'Demo XLSX product export', 'xlsx_product_export', 0, 'Akeneo Native Connector', 'a:2:{s:10:"withHeader";b:1;s:8:"filePath";s:17:"/tmp/product.xlsx";}', 'export'),
+                ('xlsx_product_quick_export', 'XLSX product quick export', 'xlsx_product_quick_export', 0, 'Akeneo Mass Edit Connector', 'a:2:{s:10:"withHeader";b:1;s:8:"filePath";s:53:"/tmp/products_export_%locale%_%scope%_%datetime%.xlsx";}', 'quick_export')
```
Comment (nidup) [inline upgrades/schema/Version_1_6_201603100000_batch_jobs.php]: No need of "xlsx_product_export", please add only "xlsx_product_quick_export". When a partner upgrades we need to add all job instances he can't add by the UI by himself (quick export and mass edit). (when a partner begin a project for a customer, he installs with a minimal data set and he adds customer data).

**Comment by nidup** [comment]
To allow to use this quick export when we install the PIM with minimal or icecat_demo_dev, you also have to add your quick export config here https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/InstallerBundle/Resources/fixtures/minimal/jobs.yml#L27 and https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/InstallerBundle/Resources/fixtures/icecat_demo_dev/jobs.yml#L216 

In EE you need to do the same and to add permissions in /PimEnterprise/Bundle/InstallerBundle/Resources/fixtures/minimal/job_profile_accesses.yml

**Comment by nidup** [comment]
Please complete the installer fixtures, fix the script then GTM

**Comment by SamirBoulil** [comment]
@nidup Alright thanks! I do that.

**Pull request merged**